src/pkg/services/m365/api/api.go

```diff
@@ -19,37 +22,87 @@ import (
 	if err != nil {
 		return Client{}, err
 	}
 
-	return Client{creds, s}, nil
+	li, err := newLargeItemService(creds)
+	if err != nil {
+		return Client{}, err
+	}
+
+	return Client{creds, s, li}, nil
 }
 
-// service generates a new service.  Used for paged and other long-running
-// requests instead of the client's stable service, so that in-flight state
-// within the adapter doesn't get clobbered
-func (c Client) Service() (*graph.Service, error) {
-	return newService(c.Credentials)
+// Service generates a new graph servicer.  New servicers are used for paged
+// and other long-running requests instead of the client's stable service,
+// so that in-flight state within the adapter doesn't get clobbered.
+// Most calls should use the Client.Stable property instead of calling this
+// func, unless it is explicitly necessary.
+func (c Client) Service() (graph.Servicer, error) {
+	return NewService(c.Credentials)
 }
 
-func newService(creds account.M365Config) (*graph.Service, error) {
-	adapter, err := graph.CreateAdapter(
+func NewService(creds account.M365Config, opts ...graph.Option) (*graph.Service, error) {
+	a, err := graph.CreateAdapter(
 		creds.AzureTenantID,
 		creds.AzureClientID,
-		creds.AzureClientSecret)
+		creds.AzureClientSecret,
+		opts...)
+	if err != nil {
+		return nil, clues.Wrap(err, "generating graph api adapter")
+	}
+
+	return graph.NewService(a), nil
+}
+
+func newLargeItemService(creds account.M365Config) (*graph.Service, error) {
+	a, err := NewService(creds, graph.NoTimeout())
 	if err != nil {
-		return nil, clues.Wrap(err, "generating graph api service client")
+		return nil, clues.Wrap(err, "generating no-timeout graph adapter")
 	}
 
-	return graph.NewService(adapter), nil
+	return a, nil
 }
+
+// ---------------------------------------------------------------------------
+// common types and consts
+// ---------------------------------------------------------------------------
+
+// DeltaUpdate holds the results of a current delta token.  It normally
+// gets produced when aggregating the addition and removal of items in
+// a delta-queryable folder.
+type DeltaUpdate struct {
+	// the deltaLink itself
+	URL string
+	// true if the old delta was marked as invalid
+	Reset bool
+}
+
+// GraphQuery represents functions which perform exchange-specific queries
+// into M365 backstore. Responses -> returned items will only contain the information
+// that is included in the options
+// TODO: use selector or path for granularity into specific folders or specific date ranges
+type GraphQuery func(ctx context.Context, userID string) (serialization.Parsable, error)
+
+// GraphRetrievalFunctions are functions from the Microsoft Graph API that retrieve
+// the default associated data of a M365 object. This varies by object. Additional
+// Queries must be run to obtain the omitted fields.
+type GraphRetrievalFunc func(
+	ctx context.Context,
+	user, m365ID string,
+) (serialization.Parsable, error)
```

Comment:
Can these actually be removed? I'm not sure if they're being used anywhere

Reply:
I'll check into that as a follow-up.  Wanted to avoid logic/removal changes in this PR.